File: EmbeddedLLMWiki/frontend/src/types/wiki.ts
export interface WikiProject {
  name: string
  path: string
}

export interface FileNode {
  name: string
  path: string
  is_dir: boolean
  children?: FileNode[]
}

export interface WikiPage {
  path: string
  content: string
  frontmatter: Record<string, unknown>
}

export interface SourceDocumentState {
  path: string
  relativePath: string
  modifiedMs: number
  size: number
}

export interface PendingIngestRequest {
  projectPath: string
  sourcePath: string
  debounceMs: number
  reason?: string
}

File: EmbeddedLLMWiki/runtime/src/types/wiki.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WikiProject {
    pub name: String,
    pub path: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FileNode {
    pub name: String,
    pub path: String,
    pub is_dir: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub children: Option<Vec<FileNode>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SourceDocumentState {
    pub path: String,
    #[serde(rename = "relativePath")]
    pub relative_path: String,
    #[serde(rename = "modifiedMs")]
    pub modified_ms: u64,
    pub size: u64,
}
